grassroots-backend/src/auth/GoogleOAuth.strategy.ts

```diff
@@ -18,31 +18,40 @@ export class GoogleOAuthStrategy extends PassportStrategy(
   GoogleStrategy,
   DEFAULT_PASSPORT_STRATEGY_NAME,
 ) {
-  constructor(
-    private config: ConfigService,
-    private userService: UsersService,
-  ) {
-    const clientID = config.get<string>("GOOGLE_CLIENT_ID");
+  constructor(private userService: UsersService) {
```

Comment:
Why the change here?

Reply:
Oh, because you can't call an async method in the constructor.

So this is a good argument for PR 3 that we were talking about (it'll let us get the awaited values dependency injected without stuff like this)

For now though, let's update the provider for the GoogleOAuth strategy.

Something like:
```
{
      provide: GoogleOAuthStrategy,
      inject: [UserService],
      useFactory: async (userService:UserService): Promise<GoogleOAuthStrategy> => {
        return new GoogleOAuthStrategy(userService, await getEnvVars());
      },
    },
```